### ISSUE TITLE
Improve WrongMagicCommentViolation rule

### ISSUE THREAD

**Issue by mirrorrim** [wemake-services/wemake-python-styleguide#476]
# Rule request

## Thesis

Allow to use `# type: some_type` in loop statements

## Reasoning

In some cases the editor cannot detect the object type.

```python
    async def iter_channel_dialogs(self) -> AsyncIterator[Dialog]:
        """Iterate through public channels."""
        async for dialog in self.iter_dialogs():  # type: Dialog
            if dialog.is_channel and dialog.entity.username:
                yield dialog
```

**Comment by sobolevn** [comment]
Hm, seams reasonable only if you can not type `self.iter_dialogs()`. Can you please show how it is typed right now?

**Comment by mirrorrim** [comment]
It is a part of Telethon library - https://github.com/LonamiWebs/Telethon/blob/master/telethon/client/dialogs.py#L15

**Comment by sobolevn** [comment]
Just a wild guess, have you tried this snippet? (I am not sure that it will work)

```python
async def iter_channel_dialogs(self) -> AsyncIterator[Dialog]:
        """Iterate through public channels."""
        dialog: Dialog
        async for dialog in self.iter_dialogs():
            ...
```

**Comment by mirrorrim** [comment]
It works, thanks. Maybe this example should be included in the docs?

**Comment by sobolevn** [comment]
@mirrorrim yes, it probably should. Please, send a PR with the fix.
Thanks for reporting this bug!

**Comment by mirrorrim** [comment]
Just opened my laptop, but the issue is already closed, did not have time :(

**Comment by sobolevn** [comment]
@mirrorrim I had some time today to work on a new release. 🙂 
No worries, take any other issue if you want to contribute! 
We have a bunch of entry-level ones opened at the moment.

**Comment by sobolevn** [comment]
@mirrorrim this issue seems like the easiest one to start with https://github.com/wemake-services/wemake-python-styleguide/issues/460

**Comment by mirrorrim** [comment]
@sobolevn thanks. I'll check how to add new rules 🙂
